src/components/NcUserBubble/NcUserBubble.vue

```diff
@@ -29,59 +29,220 @@ This component has the following slot:
+
+	/**
+	 * Whether or not to display the user-status
+	 */
+	showUserStatus?: boolean
+
+	/**
+	 * Define the whole bubble as a link
+	 */
+	url?: string
+
+	/**
+	 * Use bubble as a router-link for in-app navigation
+	 */
+	to?: RouteLocation
+
+	/**
+	 * Use the primary colour
+	 */
+	primary?: boolean
+
+	/**
+	 * This is the height of the component
+	 */
+	size?: number
+
+	/**
+	 * This is the margin of the avatar (size - margin = avatar size)
+	 */
+	margin?: number
+}>(), {
+	avatarImage: undefined,
+	displayName: undefined,
+	user: undefined,
+	url: undefined,
+	to: undefined,
+	margin: 2,
+	size: 20,
+})
+
+const emit = defineEmits<{
+	/**
+	 * The mouse click event
+	 */
+	click: [event: MouseEvent]
+}>()
+
+defineSlots<{
+	/**
+	 * Main Popover content on userbubble hover/focus
+	 */
+	default?: Slot
+
+	/**
+	 * Optional slot just after the name
+	 */
+	name?: Slot
+}>()
+
+/**
+ * Default popover state. Requires the UserBubble
+ * to have some content to render inside the popover
+ */
+const isOpen = defineModel<boolean>('open')
+
+/**
+ * Is the provided avatar url valid or not
+ */
+const isAvatarUrl = computed(() => {
+	if (!props.avatarImage) {
+		return false
+	}
+
+	try {
+		const url = new URL(props.avatarImage)
+		return !!url
+	} catch (error) {
+		return false
+	}
+})
+
+/**
+ * Do we have a custom avatar or not
+ */
+const isCustomAvatar = computed(() => !!props.avatarImage)
+
+const avatarStyle = computed(() => ({
+	marginInlineStart: `${props.margin}px`,
+}))
+
+/**
+ * Is the URL prop set
+ */
+const hasUrl = computed(() => {
+	if (!props.url || props.url.trim() === '') {
+		return false
+	}
+	try {
+		const url = new URL(props.url, props.url?.startsWith?.('/') ? window.location.href : undefined)
+		return !!url
+	} catch (error) {
+		warn('[NcUserBubble] Invalid URL passed', { url: props.url })
+		return false
+	}
+})
+
+/**
+ * href attribute to pass to content container
+ */
+const href = computed(() => hasUrl.value ? props.url : undefined)
+
+const contentComponent = computed(() => {
+	if (hasUrl.value) {
+		return 'a'
+	} else if (props.to) {
+		return RouterLink
+	} else {
+		return 'div'
+	}
+})
+
+const contentStyle = computed(() => ({
+	height: `${props.size}px`,
+	lineHeight: `${props.size}px`,
+	borderRadius: `${props.size / 2}px`,
+}))
+
+watch([() => props.displayName, () => props.user], () => {
+	if (!props.displayName && !props.user) {
+		warn('[NcUserBubble] At least `displayName` or `user` property should be set.')
+	}
+})
+</script>
+
 <template>
-	<component :is="!!$slots.default ? 'NcPopover' : 'NcUserBubbleDiv'"
-		trigger="hover focus"
-		:shown="open"
+	<component :is="!!$slots.default ? NcPopover : NcUserBubbleDiv"
+		v-model:shown="isOpen"
 		class="user-bubble__wrapper"
-		@update:open="onOpenChange">
+		trigger="hover focus">
 		<!-- Main userbubble structure -->
 		<template #trigger="{ attrs }">
-			<component :is="isLinkComponent"
+			<component :is="contentComponent"
```

Comment:
You didn't like the name `isLinkComponent`? :P

Reply:
No its not a boolean, confused me while reading (usually `isXXX` is used for boolean getters or states), IMHO its more clear with that name that it is the component used for the content (maybe also contentTagName or similar).

Reply:
Sorry, it wasn't a serious question 😀 
`*"sarcasm" sign*`